packages/core/strapi/lib/services/entity-service/types/result.d.ts

```diff
@@ -36,3 +75,133 @@ type GetValue<TAttribute extends Attribute.Attribute> = Utils.Expression.If<
+      ParseStringFields<
+        TSchemaUID,
+        As<TFields, EntityService.Params.Fields.StringNotation<TSchemaUID>>
+      >
+    ],
+    // string array
+    [
+      Utils.Expression.Extends<TFields, EntityService.Params.Fields.ArrayNotation<TSchemaUID>>,
+      ParseStringFields<
+        TSchemaUID,
+        As<
+          Utils.Array.Values<As<TFields, EntityService.Params.Fields.ArrayNotation<TSchemaUID>>>,
+          EntityService.Params.Fields.StringNotation<TSchemaUID>
+        >
+      >
+    ]
+  ]
+>;
+
+type ParseStringFields<
+  TSchemaUID extends Common.UID.Schema,
+  TFields extends EntityService.Params.Fields.StringNotation<TSchemaUID>
+> = Utils.Expression.MatchFirst<
+  [
+    [Utils.Expression.IsNever<TFields>, Attribute.GetNonPopulatableKeys<TSchemaUID>],
+    [
+      Utils.Expression.Extends<EntityService.Params.Fields.WildcardNotation, TFields>,
+      Attribute.GetNonPopulatableKeys<TSchemaUID>
+    ],
+    [
+      Utils.Expression.Extends<TFields, EntityService.Params.Fields.SingleAttribute<TSchemaUID>>,
+      TFields
+    ],
+    [
+      // what aobut * in comma separated list
+      Utils.Expression.Extends<TFields, `${string},${string}`>,
+      Utils.Array.Values<Utils.String.Split<As<TFields, string>, ','>>
+    ]
+  ]
+>;
+
+type ExtractPopulate<
+  TSchemaUID extends Common.UID.Schema,
+  TPopulate extends EntityService.Params.Populate.Any<TSchemaUID>
+> = Utils.Expression.MatchFirst<
+  [
+    [
+      Utils.Expression.Extends<TPopulate, EntityService.Params.Populate.StringNotation<TSchemaUID>>,
+      ParseStringPopulate<
+        TSchemaUID,
+        As<TPopulate, EntityService.Params.Populate.StringNotation<TSchemaUID>>
+      >
+    ],
+    [
+      Utils.Expression.Extends<TPopulate, EntityService.Params.Populate.ArrayNotation<TSchemaUID>>,
+      ParseStringPopulate<
+        TSchemaUID,
+        As<
+          Utils.Array.Values<
+            As<TPopulate, EntityService.Params.Populate.ArrayNotation<TSchemaUID>>
+          >,
+          EntityService.Params.Populate.StringNotation<TSchemaUID>
+        >
+      >
+    ],
+    [Utils.Expression.Extends<Object, TPopulate>, never],
+    [
+      Utils.Expression.Extends<TPopulate, EntityService.Params.Populate.ObjectNotation<TSchemaUID>>,
+      ParseStringPopulate<
+        TSchemaUID,
+        As<keyof TPopulate, EntityService.Params.Populate.StringNotation<TSchemaUID>>
+      >
+    ]
+  ],
+  never
+>;
+
+type ParseStringPopulate<
+  TSchemaUID extends Common.UID.Schema,
+  TPopulate extends EntityService.Params.Populate.StringNotation<TSchemaUID>
+> = Utils.Expression.MatchFirst<
+  [
+    [Utils.Expression.IsNever<TPopulate>, Attribute.GetPopulatableKeys<TSchemaUID>],
+    [
+      Utils.Expression.Extends<EntityService.Params.Populate.WildcardNotation, TPopulate>,
+      Attribute.GetPopulatableKeys<TSchemaUID>
+    ],
+    [
+      // what aobut * in comma separated list
+      Utils.Expression.Extends<TPopulate, `${string},${string}`>,
```

Comment:
We also have `${string}.${string}` to take into account here (both are valid for populate strings)